docs/changes/io.ascii/15496.api.rst

```diff
@@ -0,0 +1 @@
+Removed usage of ``OrderedDict`` from several files within the ``io.ascii`` subpackage. Affected files include: ``core.py``, ``daophot.py``, ``ecsv.py``, ``fastbasic.py``, ``ipac.py``, and ``test_read.py``. With the guaranteed insertion order in dictionaries from Python 3.7 onwards, the shift to standard dictionaries simplifies the code and reduces reliance on specialized data structures. This refactoring aligns with issue #10888.
```

Comment:
```suggestion
Replaced ``OrderedDict`` instances within ``io.ascii`` with standard dictionaries, which preserve insertion order as of Python 3.7. Changed calls to ``yaml.dump`` to default to ``sort_keys=False`` to ensure this order is preserved on serialization as well.
```
No need to list all modified files here, as they will be easily identifiable by `git log --name-status`. Internal uses of dict/OrderedDict may not require a changelog at all, but if this finally succeeds with the modified dump, it should probably have one, as it will output different yaml files for standard dictionaries.

@pllim : the other PRs for `io` were merged w/o changelog, but #15491 has one for API change; not quite sure if this really constitutes a change in API (see above).

Reply:
Thanks for your advice and patience!

Reply:
Thank you for your persistence on what turns out to be a far from straightforward refactor!
I just committed the change directly since the tests had been stuck on the previous commit somehow; unfortunately we are still not quite there.